src/mbgl/renderer/layers/render_line_layer.cpp

```diff
@@ -214,14 +214,16 @@ void RenderLineLayer::render(PaintParameters& parameters) {
     }
 }
 
-optional<GeometryCollection> offsetLine(const GeometryCollection& rings, const double offset) {
-    if (offset == 0) return {};
+std::unique_ptr<GeometryCollection> offsetLine(const GeometryCollection& rings, const double offset) {
+
+    if (offset == 0) return nullptr;
+
+    std::unique_ptr<GeometryCollection> newRings = std::make_unique<GeometryCollection>();
```

Comment:
After some checks I came to conclusion that we could keep the old semantics with the `optional` return type.
The problem was caused by implicit copy of the local variable on return, that can be fixed just with adding `std::move()` call (RVO is failing here as local and return types are different).
```
optional<GeometryCollection> offsetLine() {
GeometryCollection newRings;
return newRings; // Copy here!
}

optional<GeometryCollection> offsetLine() {
GeometryCollection newRings;
return std::move(newRings); // No copy :)
}
```

Reply:
Gotcha - will test out.

Reply:
`return std::move(newRings);` will prevent copy elision, I think we even have clang check for that.

Another option is to `return optional<GeometryCollection>(std::move(newRings));`

Reply:
hm, another issue could be that we have two return statements that could prevent RVO.

Reply:
> return std::move(newRings); will prevent copy elision

in this case we're creating an instance of another type:  `optional<GeometryCollection>`. Returning rvalue triggers converting move constructor; returning lvalue triggers converting copy constructor.